### ISSUE TITLE
Bad forwarding from >1.14.2

### ISSUE THREAD

**Issue by zifeo** [txn2/kubefwd#150]
The upgrade from 1.14.2 to 1.14.7 does not handle statefulset as expected.

1.14.2
```
INFO[03:21:36] Port-Forward: db-replicaset-client:27017 to pod db-replicaset-0:27017 
INFO[03:21:36] Port-Forward: db-replicaset-0.db-replicaset-client:27017 to pod db-replicaset-0:27017 
INFO[03:21:36] Port-Forward: db-replicaset-1.db-replicaset-client:27017 to pod db-replicaset-1:27017 
INFO[03:21:36] Port-Forward: db-replicaset-2.db-replicaset-client:27017 to pod db-replicaset-2:27017 
INFO[03:21:36] Port-Forward: db-replicaset:27017 to pod db-replicaset-0:27017 
INFO[03:21:36] Port-Forward: db-replicaset-0.db-replicaset:27017 to pod db-replicaset-0:27017 
INFO[03:21:36] Port-Forward: db-replicaset-1.db-replicaset:27017 to pod db-replicaset-1:27017 
INFO[03:21:36] Port-Forward: db-replicaset-2.db-replicaset:27017 to pod db-replicaset-2:27017 
```

1.14.7
```
INFO[03:20:34] Port-Forward: db-replicaset-client:27017 to pod db-replicaset-0:27017 
INFO[03:20:34] Port-Forward: db-replicaset-client:27017 to pod db-replicaset-0:27017 
INFO[03:20:34] Port-Forward: db-replicaset-client:27017 to pod db-replicaset-1:27017 
INFO[03:20:34] Port-Forward: db-replicaset-client:27017 to pod db-replicaset-2:27017 
INFO[03:20:34] Port-Forward: db-replicaset:27017 to pod db-replicaset-0:27017 
INFO[03:20:34] Port-Forward: db-replicaset:27017 to pod db-replicaset-0:27017 
INFO[03:20:34] Port-Forward: db-replicaset:27017 to pod db-replicaset-1:27017 
INFO[03:20:34] Port-Forward: db-replicaset:27017 to pod db-replicaset-2:27017 
```

Brew rollback `wget https://raw.githubusercontent.com/txn2/homebrew-tap/fb0bc557ada8de65a1577565166e006ef76ff3c8/Formula/kubefwd.rb && brew reinstall kubefwd.rb && rm kubefwd.rb` (`brew pin kubefwd` when upgrading, do not forget to unpin later).

Relates to https://github.com/txn2/kubefwd/issues/133 & https://github.com/txn2/kubefwd/pull/134.

**Comment by cjimti** [comment]
Please try the new 1.17.0. Thanks!

**Comment by zifeo** [comment]
@cjimti thanks, while the log seems fine the `/etc/hosts` seems not corrected (the pods get listed twice):
```
127.1.27.3       db-replicaset-0.db-replicaset-0.db-replicaset
```

**Comment by cjimti** [comment]
I am sorry to see that. I have not seen this in my testing. Can you post the service configuration, your OS, etc? I would like to try to re-produce this.

**Comment by zifeo** [comment]
@cjimti MacOS (latest) and the service is configured as follows on GKE 1.17:
```
kind: Service
metadata:
  annotations:
    meta.helm.sh/release-name: db
    meta.helm.sh/release-namespace: db-namespace
    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
  creationTimestamp: "2020-02-16T15:48:39Z"
  labels:
    app: db-replicaset
    app.kubernetes.io/managed-by: Helm
    chart: db-replicaset-2.15.1
    heritage: Helm
    release: XXX
  name: db-replicaset
  namespace: db-namespace
  resourceVersion: "1765653605"
  selfLink: /api/v1/namespaces/mongo/services/db-replicaset
  uid: 5433202c-3202-11e9-8f75-48f75a3202a5
spec:
  clusterIP: None
  ports:
  - name: db
    port: 2017
    protocol: TCP
    targetPort: 2017
  publishNotReadyAddresses: true
  selector:
    app: db-replicaset
    release: XXX
  sessionAffinity: None
  type: ClusterIP
```

**Comment by zifeo** [comment]
@cjimti should the issue be re-opened?

**Comment by cjimti** [comment]
I believe I found the issue and fixed it in 1.7.2

**Comment by zifeo** [comment]
@cjimti I confirm 👍  thanks
